src/utils/resources/vm/hooks/useVMIForVM.ts

```diff
@@ -0,0 +1,29 @@
+import { VirtualMachineInstanceModelGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseVMIForVM = (
+  vmName: string,
+  vmNamespace: string,
+) => {
+  vmi: V1VirtualMachineInstance;
+  vmiLoaded: boolean;
+  vmiLoadError: any;
+};
+
+const useVMIForVM: UseVMIForVM = (vmName, vmNamespace) => {
+  const [vmi, vmiLoaded, vmiLoadError] = useK8sWatchResource<V1VirtualMachineInstance>({
+    groupVersionKind: VirtualMachineInstanceModelGroupVersionKind,
+    isList: false,
+    name: vmName,
+    namespace: vmNamespace,
+  });
+
+  return {
+    vmi: vmName === vmi?.metadata?.name && vmNamespace === vmi?.metadata?.namespace && vmi,
```

Comment:
i dont think its necessary as u already sending the name and namespace in the request.. 
btw i recommend the name of the hook to be something like useGetVMI (as this is what it does)
it can also get the VM instead of 2 args and just extract here from metadata

Reply:
Why not just `useVMI`? The `Get` makes no sense to me with the prefix `use`. I was only using for the `forVM` convention because it's what was being used already.

This hook is use by `useVMIAndPodsForVM` which doesn't have access to the VM. I check its usages and there are components that use it that only have the VM name and namespace that they get from URL parameters.

Reply:
the use is mandatory by react to be declared as a hook. other wise we will remove it also :) 

the get simple indicates that its only fetches the vmi, but useVMI can work also.


> This hook is use by useVMIAndPodsForVM which doesn't have access to the VM. I check its usages and there are components that use it that only have the VM name and namespace that they get from URL parameters.

ok